sabnzbd/filesystem.py

```diff
@@ -1244,3 +1245,35 @@ def save_compressed(folder: str, filename: str, data_fp: BinaryIO) -> str:
         logging.info("Skipping existing file %s", full_nzb_path)
 
     return full_nzb_path
+
+
+def directory_is_writable(mydir, myfilename):
+    filename = os.path.join(mydir, myfilename)
+    if os.path.exists(filename):
+        try:
+            os.remove(filename)
+        except:
+            return False
+    try:
+        with open(filename, "w") as f:
+            f.write("Some random content")
+        os.remove(filename)
+        return True
+    except:
+        return False
+
+
+def check_directory_writing_capability(dir):
+    """Checks if dir is writable at all, and writable with special chars. Returns Error, Message"""
+    SOME_WRITING_ERROR = True
+    if directory_is_writable(dir, "sab_test.txt"):
+        if not sabnzbd.WIN32 and not sabnzbd.cfg.sanitize_safe():
+            if not directory_is_writable(dir, "sab_test \\ bla :: , bla.txt"):
+                return (
```

Comment:
Let's just use `True` and `False` as return variables.
You can call `helpful_warning` in `filesystem.py` by calling `sabnzbd.misc.helpful_warning(T('..'))`

Reply:
OK, but if we do the helpful_warning (and thus printing) in filesystem, is there still a return value needed? Maybe only for the unit_test?